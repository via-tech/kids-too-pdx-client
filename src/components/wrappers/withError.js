import React from 'react';
import PropTypes from 'prop-types';
import styles from './Wrappers.css';

export const withError = Component => {
  function WithError(props) {
    const { error } = props;

    return (
      <div className={styles.Wrappers}>
        {error && <p className={styles.Error}>{error}</p>}

        <Component {...props} />
      </div>
    );
  }

  WithError.propTypes = {
    error: PropTypes.string
  };

  return WithError;
};
