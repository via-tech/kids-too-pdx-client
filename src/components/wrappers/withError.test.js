import React from 'react';
import renderer from 'react-test-renderer';
import { withError } from './withError';

describe('With error wrapper', () => {
  it('snapshots withError with error', () => {
    const props = {
      error: 'Wrong!!'
    };

    const Component = withError(() =>
      <div>
        <h1>The Main Component</h1>
      </div>
    );

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });

  it('snapshots withError without error', () => {
    const props = {
      error: ''
    };

    const Component = withError(() =>
      <div>
        <h1>The Main Component</h1>
      </div>
    );

    const tree = renderer.create(
      <Component {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
