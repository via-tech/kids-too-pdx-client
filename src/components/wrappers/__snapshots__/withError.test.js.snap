// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With error wrapper snapshots withError with error 1`] = `
<div
  className="Wrappers"
>
  <p
    className="Error"
  >
    Wrong!!
  </p>
  <div>
    <h1>
      The Main Component
    </h1>
  </div>
</div>
`;

exports[`With error wrapper snapshots withError without error 1`] = `
<div
  className="Wrappers"
>
  
  <div>
    <h1>
      The Main Component
    </h1>
  </div>
</div>
`;
